pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -928,14 +960,25 @@ func (vs *VSphere) HasClusterID() bool {
 	return true
 }
 
-// Notification handler when node is registered.
-func (vs *VSphere) NodeRegistered(node *v1.Node) {
-	glog.V(4).Infof("Node Registered: %+v", node)
+// Notification handler when node is added into k8s cluster.
+func (vs *VSphere) NodeAdded(obj interface{}) {
+	node, ok := obj.(*v1.Node)
+	if node == nil || !ok {
+		return
```

Comment:
When can this fail? This is a crucial decision since we are not doing anything for this notification. Could you please a log that we are ignoring a notification for this obj?

Reply:
Sure.